game_session.js

```diff
@@ -63,4 +66,48 @@ module.exports = class GameSession {
     getDefaultBeginningCutoffYear() {
         return BEGINNING_SEARCH_YEAR;
     }
+
+    resetGender() {
+        this._gameOptions.gender = [ FEMALE ];
+    }
+
+    setGender(genderArr) {
+        // Return true when gender is successfully updated, false otherwise
+        if (genderArr.length === 0) {
+            this.resetGender();
+        }
+        else {
+            genderArr.map((gender, i) => { genderArr[i] = genderArr[i].toLowerCase(); })
+            this._gameOptions.gender = [];
+            for (let i = 0; i < genderArr.length; i++) {
+                if (!this._gameOptions.gender.includes(genderArr[i]) &&
```

Comment:
Can replace this with `!this._gameOptions.gender.includes(genderArr[i]) && Object.keys(GENDER).includes(genderArr[i])`

Reply:
this probably works too
```
this._gameOptions = genderArr.filter((gender, pos) => { 
     gender = gender.toLowerCase();
     let firstOccurence = genderArr.indexOf(gender) == pos;
     return firstOccurence && Object.keys(GENDER).includes(gender);
})
```